app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -86,6 +86,7 @@ def inspect_work
 
       # Overridden to provide depositor (used on app/views/hyrax/base/_form_share.html.erb)
       def build_change_set(resource)
+        raise Hydra::AccessDenied unless current_user
```

Comment:
This method is about building a change set, not about checking the users access, so is there a different place we could put this?

Reply:
Would flipping lines 15 & 16 do the trick? https://github.com/samvera/hyrax/blob/204e6b6d529daaff9d6eb25c596dfade91d87a5d/app/controllers/concerns/hyrax/resource_controller.rb#L16

Reply:
Yes, that does it. Thanks!